types/src/front/data_source_view.ts

```diff
@@ -17,13 +17,16 @@ export interface DataSourceViewType {
   parentsIn: string[] | null;
   sId: string;
   updatedAt: number;
-  // TODO(GROUPS_INFRA) Add support for usage.
-  usage: number;
   vaultId: string;
 }
 
-export type DataSourceViewWithConnectorType = DataSourceViewType & {
-  dataSource: DataSourceWithConnectorDetailsType;
+export type DataSourceViewsWithDetails = DataSourceViewType & {
+  dataSource: DataSourceType & {
+    connector: ConnectorType | null;
+    fetchConnectorError: boolean;
+    fetchConnectorErrorMessage: string | null;
+  };
```

Comment:
Why did we get rid of `DataSourceWithConnectorDetailsType`? Especially if we using the exact same type? Or let's create a type for:
```
{
    connector: ConnectorType | null;
    fetchConnectorError: boolean;
    fetchConnectorErrorMessage: string | null;
  }
```
so that we can reuse it.

Reply:
It's not the exact same type actually :)

DataSourceWithConnectorDetailsType enforce WithConnector who disable "null" connectors.
However, in the endpoint for datasourceviews with details, we return managed and non managed sources (where connector is null).

> Or let's create a type for

Will do !